### PR TITLE
Date boundary issue

### PR DESCRIPTION
**Pull request by YangYumings**
Modify four places
- Use short type as the parameter for converting LocalDate to prevent crossing the range [1000, 9999];
- Set the thread that records configuration information as a daemon thread;
   NamedThreadFactory can optionally set the thread as a daemon thread
- <del>Add the -fn (file name) parameter on the original basis. It is used to specify the name of the configuration file. It can be used alone.</del>
- Solve the problem that the number of created tables is less than IoTDB_TABLE NUMBER.

### CODE REVIEW

File: configuration/bin/startup.sh [hunk 1]
```diff
@@ -44,6 +45,11 @@ while [[ $# -gt 0 ]]; do
       shift
       shift
       ;;
+    -fn)
+      config_filename="$2"
+      shift
+      shift
+      ;;
     -maxheapsize)
       MAX_HEAP_SIZE="$2"
```
Comment (SpriCoder) [inline configuration/bin/startup.sh:54]: the params after config_filename need to update ${number}

Comment (SpriCoder) [reply configuration/bin/startup.sh:54]: eg. MAX_HEAP_SIZE="$3"

File: configuration/test/conf/config.properties [hunk 2]
```diff
@@ -0,0 +1,485 @@
+########################################################
```
Comment (SpriCoder) [inline configuration/test/conf/config.properties]: why add test/conf/properties here?

**Review by SpriCoder — CHANGES_REQUESTED** [review]
PTAL~

File: configuration/bin/startup.sh (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply configuration/bin/startup.sh:54]: same

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java [hunk 3]
```diff
@@ -75,14 +75,17 @@ public Config getConfig() {
...
+      builder
+          .append("/")
+          .append(System.getProperty(Constants.BENCHMARK_CONF_FILENAME, "config.properties"));
       InputStream inputStream;
       try {
-        inputStream = new FileInputStream(url);
+        inputStream = new FileInputStream(builder.toString());
       } catch (FileNotFoundException e) {
-        LOGGER.warn("Fail to find config file {}", url);
+        LOGGER.warn("Fail to find config file {}", builder);
```
Comment (OneSizeFitsQuorum) [inline core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java:88]: 前面统一 builder.toString()?

File: configuration/test/conf/config.properties (hunk 2 above)
Comment (YangYumings) [reply configuration/test/conf/config.properties]: This test file has been deleted

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java (hunk 3 above)
Comment (YangYumings) [reply core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java:88]: ok

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~

关于那个 BM 无法正常退出的问题，搞下一个 PR 也行，或者找到了在这个 PR 里面改了也行。

**Review by SpriCoder — APPROVED** [review]
Only one minor suggestions PTAL

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/utils/NamedThreadFactory.java [hunk 4]
```diff
@@ -22,6 +29,9 @@ private String getThreadName() {
   public Thread newThread(Runnable r) {
     Thread thread = threadFactory.newThread(r);
     thread.setName(getThreadName());
+    if (daemon) {
+      thread.setDaemon(true);
+    }
```
[the remark this answers is not in the document]
Comment (YangYumings) [reply core/src/main/java/cn/edu/tsinghua/iot/benchmark/utils/NamedThreadFactory.java]: Modified.

**Pull request merged**